src/schemes/fluid/surface_normal_sph.jl

```diff
@@ -0,0 +1,144 @@
+    surface_normal = Array{ELTYPE, 2}(undef, NDIMS, nparticles)
+    neighbor_count = Array{ELTYPE, 1}(undef, nparticles)
+    return (; surface_normal, neighbor_count)
+end
+
+@inline function surface_normal(particle_system::FluidSystem, particle)
+    (; cache) = particle_system
+    return extract_svector(cache.surface_normal, particle_system, particle)
+end
+
+# Section 2.2 in Akinci et al. 2013 "Versatile Surface Tension and Adhesion for SPH Fluids"
+# Note: This is the simplest form of normal approximation commonly used in SPH and comes
+# with serious deficits in accuracy especially at corners, small neighborhoods and boundaries
+function calc_normal_akinci!(system, neighbor_system::FluidSystem, u_system, v,
+                             v_neighbor_system, u_neighbor_system, semi, surfn)
+    (; cache) = system
+
+    system_coords = current_coordinates(u_system, system)
+    neighbor_system_coords = current_coordinates(u_neighbor_system, neighbor_system)
+    nhs = get_neighborhood_search(system, neighbor_system, semi)
+
+    foreach_point_neighbor(system, neighbor_system,
+                           system_coords, neighbor_system_coords,
+                           nhs) do particle, neighbor, pos_diff, distance
+        m_b = hydrodynamic_mass(neighbor_system, neighbor)
+        density_neighbor = particle_density(v_neighbor_system,
+                                            neighbor_system, neighbor)
+        grad_kernel = smoothing_kernel_grad(system, pos_diff, distance)
+        for i in 1:ndims(system)
+            cache.surface_normal[i, particle] += m_b / density_neighbor * grad_kernel[i]
+        end
+
+        cache.neighbor_count[particle] += 1
+    end
+
+    return system
+end
+
+# Section 2.2 in Akinci et al. 2013 "Versatile Surface Tension and Adhesion for SPH Fluids"
+# Note: This is the simplest form of normal approximation commonly used in SPH and comes
+# with serious deficits in accuracy especially at corners, small neighborhoods and boundaries
+function calc_normal_akinci!(system, neighbor_system::BoundarySystem, u_system, v,
+                             v_neighbor_system, u_neighbor_system, semi, surfn)
+    (; cache) = system
+    (; colorfield, colorfield_bnd) = neighbor_system.boundary_model.cache
+
+    system_coords = current_coordinates(u_system, system)
+    neighbor_system_coords = current_coordinates(u_neighbor_system, neighbor_system)
+    nhs = get_neighborhood_search(system, neighbor_system, semi)
+
+    # First we need to calculate the smoothed colorfield values
+    # TODO: move colorfield to extra step
+    # TODO: this is only correct for a single fluid
+    set_zero!(colorfield)
+
+    foreach_point_neighbor(system, neighbor_system,
+                           system_coords, neighbor_system_coords,
+                           nhs) do particle, neighbor, pos_diff, distance
+        colorfield[neighbor] += smoothing_kernel(system, distance)
+    end
```

Comment:
```suggestion
    foreach_point_neighbor(neighbor_system, system,
                           neighbor_system_coords, system_coords,
                           nhs) do particle, neighbor, pos_diff, distance
        colorfield[particle] += smoothing_kernel(system, distance)
    end
```
Otherwise you will get race conditions on multiple threads.

Reply:
Doesn't work there will be no neighbors

Reply:
`foreach_point_neighbor(neighbor_system, system, ..., particles=eachparticle(neighbor_system))`, or the loop will go over all moving particles, which are none for a boundary system.